code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/ServiceDeploymentPlanner.java

```diff
@@ -28,7 +28,7 @@ public ServiceDeploymentPlanner(List<Service> services, List<DeploymentUnit> uni
 
         if (units != null) {
             for (DeploymentUnit unit : units) {
-                if (unit.isUnhealthy()) {
+                if (unit.isUnhealthy() || !unit.isHostActive()) {
```

Comment:
Son, is there a reason why we segregate "isHostActive" to its separate method, instead of making it a part of isUnhealthy? Because its really what it means - unit is unhealthy when the healtcheck failed for it, or when the host where unit runs on, is unhealthy. I think it would make code cleaner if we don't expose TMI to the deployment manager.